examples/twittermap/web/public/javascripts/searchbar/controllers.js

```diff
@@ -4,7 +4,64 @@ angular.module('cloudberry.util', ['cloudberry.common'])
       if ($scope.keyword && $scope.keyword.trim().length > 0) {
         cloudberry.parameters.keywords = $scope.keyword.trim().split(/\s+/);
         // skip the empty query for now.
-        cloudberry.query(cloudberry.parameters);
+
+          //stopword filtering feature
+          String.prototype.removeStopWords = function() {
+              var x;
+              var y;
+              var word;
+              var stop_word;
+              var regex_str;
+              var regex;
+              var cleansed_string = this.valueOf();
+              var stop_words = external_stop_words();
+
+              // Split out all the individual words in the phrase
+              var words = cleansed_string.match(/[^\s]+|\s+[^\s+]$/g);
+
+              // Review all the words
+              for(x=0; x < words.length; x++) {
+                  // For each word, check all the stop words
+                  for(y=0; y < stop_words.length; y++) {
```

Comment:
This works. But, it's slow since for each word, you iterate all stopwords. What I have proposed in the issue is:

1. Build a hashmap from stopwords.
2. Tokenize the query into multiple tokens.
3. For each token, check if the hashmap contains it. If so, remove it from tokens array.
4. Build the final query from tokens array and send it to the cloudberry.